py-polars/tests/unit/operations/test_inefficient_apply.py

```diff
@@ -31,15 +31,20 @@ def _get_suggestion(
         np.sin,
         lambda x: np.sin(x),
         lambda x, y: x + y,
-        lambda x: MY_CONSTANT + x,
         lambda x: x[0] + 1,
         lambda x: x,
         lambda x: x > 0 and (x < 100 or (x % 2 == 0)),
+        lambda x: MY_CONSTANT + 42,
     ],
 )
 def test_non_simple_function(func: Callable[[Any], Any]) -> None:
-    assert not _param_name_from_signature(func) or not _can_rewrite_as_expression(
-        _get_bytecode_instructions(func), apply_target="expr"
+    assert (
+        not _param_name_from_signature(func)
+        or not _can_rewrite_as_expression(
+            _get_bytecode_instructions(func), apply_target="expr"
+        )
+        or _get_suggestion(func, col="colx", apply_target="expr", param_name="x")
+        is not None
```

Comment:
bit confused by this condition - aren't we checking that there don't warn / suggest? checking that the suggestion is not None means checking that there is a suggestion?

Reply:
I just moved it to a separate test - it's confirming that we don't generate a warning if the function implies a const result (same as when we disallow `lambda x: 42`).